src/pool-cl/lens/TickLens.sol

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity 0.8.26;
+
+import {PoolKey} from "pancake-v4-core/src/types/PoolKey.sol";
+import {PoolId, PoolIdLibrary} from "pancake-v4-core/src/types/PoolId.sol";
+import {ICLPoolManager} from "pancake-v4-core/src/pool-cl/interfaces/ICLPoolManager.sol";
+import {CLPoolParametersHelper} from "pancake-v4-core/src/pool-cl/libraries/CLPoolParametersHelper.sol";
+import {Tick} from "pancake-v4-core/src/pool-cl/libraries/Tick.sol";
+import {TickMath} from "pancake-v4-core/src/pool-cl/libraries/TickMath.sol";
+import {ITickLens} from "../interfaces/ITickLens.sol";
+
+/// @title Tick Lens contract
+contract TickLens is ITickLens {
+    using PoolIdLibrary for PoolId;
+    using CLPoolParametersHelper for bytes32;
+
+    ICLPoolManager public immutable poolManager;
+
+    constructor(ICLPoolManager _poolManager) {
+        poolManager = _poolManager;
+    }
+
+    /// @inheritdoc ITickLens
+    function getPopulatedTicksInWord(PoolKey memory key, int16 tickBitmapIndex)
+        external
+        view
+        override
+        returns (PopulatedTick[] memory populatedTicks)
+    {
+        PoolId id = key.toId();
+        int24 tickSpacing = key.parameters.getTickSpacing();
+        return getPopulatedTicksInWord(id, tickSpacing, tickBitmapIndex);
+    }
+
+    /// @inheritdoc ITickLens
+    function getPopulatedTicksInWord(PoolId id, int24 tickSpacing, int16 tickBitmapIndex)
```

Comment:
since we stored `poolManager`, any chance we can make this interface easier for FE? eg.

```
PoolId id, int24 tickSpacing, int16 tickBitmapIndex

to

PoolId id, int16 tickBitmapIndex
```


**How**

PoolKey memory key  = clPoolManager.poolIdToPoolKey(poolId)
int24 tickSpacing = key.parameters.getTickSpacing();

Reply:
ah, its fine. i will make the change locally on my side and submit PR (i have some changes locally)

Reply:
actually at first i think FE only need to use "getPopulatedTicksInWord(PoolKey memory key, int16 tickBitmapIndex)", should be fine.
getPopulatedTicksInWord(PoolId id, int24 tickSpacing, int16 tickBitmapIndex) just want to give more options.

Reply:
added: https://github.com/pancakeswap/pancake-v4-periphery/pull/40/files can see if its fine (no worries if you feel that PR doesn't help, can close it)